brushfire-core/src/main/scala/com/stripe/brushfire/Brushfire.scala

```diff
@@ -46,10 +46,9 @@ trait Split[V, T] {
   def predicates: Iterable[(Predicate[V], T)]
 }
 
-/** Evaluates the goodness of a candidate split */
-trait Evaluator[V, T] {
-  /** returns a (possibly transformed) version of the input split, and a numeric goodness score */
-  def evaluate(split: Split[V, T]): (Split[V, T], Double)
+trait Evaluator[T] {
+  /** returns an overall numeric training error for a tree or split, or None for infinite/unacceptable error */
+  def trainingError(root: T, leaves: Iterable[T]): Option[Double]
```

Comment:
I also wonder whether the `Option[T]` is actually confounding two concepts: "unacceptable" should really be something that `Stopper` determines separately. That is, `Evaluator` would always return `Double` but `Stopper` would include both  `isWorthTryingToSplit` and `isAcceptableSplitOutcome`.